### PR TITLE
feat: enable taiko

### PR DESCRIPTION
**Pull request by myz1237**
https://lifi.atlassian.net/browse/LF-9267

### CODE REVIEW

File: src/chains/supportedChains.evm.ts [hunk 1]
```diff
@@ -899,4 +899,32 @@ export const supportedEVMChains: EVMChain[] = [
...
+    coin: CoinKey.ETH,
+    id: 167000,
+    mainnet: true,
+    logoURI:
+      'https://raw.githubusercontent.com/lifinance/types/main/src/assets/icons/chains/taiko.png',
+    multicallAddress: multicallAddresses[ChainId.TAI],
+    metamask: {
+      chainId: prefixChainId(167000),
+      blockExplorerUrls: ['https://taikoscan.io/'],
+      chainName: 'Taiko',
```
[the remark this answers is not in the document]
Comment (myz1237) [reply src/chains/supportedChains.evm.ts:917]: thanks for your reminder

[the remark this answers is not in the document]
Comment (myz1237) [reply src/chains/supportedChains.evm.ts:917]: what is MM warnings?

[the remark this answers is not in the document]
Comment (chybisov) [reply src/chains/supportedChains.evm.ts:917]: > what is MM warnings?

When you add a chain to MM if it has a wrong chain name or RPC URL you can have these warnings:

image

So, when adding new chains, we need to ensure the chain name and default RPC URL (the first one in the list) are among the allowed by MM. We fixed that recently here https://github.com/lifinance/data-types/pull/54

[the remark this answers is not in the document]
Comment (myz1237) [reply src/chains/supportedChains.evm.ts]: But the problem is that it shows this warning as well, when I added network to mm.
image

The question is that how to know what information mm thinks legal and how to find it

[the remark this answers is not in the document]
Comment (chybisov) [reply src/chains/supportedChains.evm.ts]: @myz1237 usually you have a button somewhere on related to the chain websites like https://taikoscan.io/

image

image



Please bring back the RPC URLs you removed and put https://rpc.mainnet.taiko.xyz as first 😁

[the remark this answers is not in the document]
Comment (myz1237) [reply src/chains/supportedChains.evm.ts]: ok, I see. Magic. wil take care of it next time

**Review by chybisov — APPROVED** [review]

File: src/chains/supportedChains.evm.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (myz1237) [reply src/chains/supportedChains.evm.ts]: will the test throw an error if I am not using rpc verified by mm?

[the remark this answers is not in the document]
Comment (chybisov) [reply src/chains/supportedChains.evm.ts]: > will the test throw an error if I am not using rpc verified by mm?

There is no such test yet, please, you are free to add one.

**Pull request merged**